compiler/src/iree/compiler/Codegen/Common/SpecializeExports.cpp

```diff
@@ -0,0 +1,463 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cstdint>
+#include <numeric>
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "iree/compiler/Dialect/TensorExt/IR/TensorExtOps.h"
+#include "iree/compiler/Dialect/Util/IR/UtilOps.h"
+#include "iree/compiler/Dialect/Util/IR/UtilTypes.h"
+#include "llvm/ADT/SlowDynamicAPInt.h"
+#include "llvm/Support/Casting.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/Visitors.h"
+
+#define DEBUG_TYPE "iree-codegen-specialize-exports"
+
+using RangeAttrHelper = mlir::iree_compiler::IREE::Codegen::IREECodegenDialect::
```

Comment:
style: usings aren't great for readability - if using them (hah) then don't rename types - someone searching the codebase for `SpecializationRangesAttrHelper` is not going to find your uses in here, and someone landing in this code is going to see `RangeAttrHelper` and not know it is `SpecializationRangesAttrHelper`.

Reply:
I normally avoid `using` but the typename was very long here. Changed to keep the name the same (but it's still technically an alias because the helper is a class member).